Mathlib/Algebra/BigOperators/WithTop.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2024 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou, Yaël Dillies
+-/
+import Mathlib.Algebra.BigOperators.Basic
+import Mathlib.Algebra.Order.Ring.WithTop
+
+/-!
+# Sums in `WithTop`
+
+This file proves results about finite sums over monoids extended by a bottom or top element.
+-/
+
+open Finset
+open scoped BigOperators
+
+variable {ι α : Type*}
+
+namespace WithTop
+section AddCommMonoid
+variable [AddCommMonoid α] [LT α] {s : Finset ι} {f : ι → WithTop α}
+
+@[simp, norm_cast] lemma coe_sum (s : Finset ι) (f : ι → α) :
+    ∑ i in s, f i = ∑ i in s, (f i : WithTop α) := map_sum addHom f s
```

Comment:
Not an issue per se, but it's not obvious simply by looking at the code that the lhs is coerced to `WithTop α`, since usually the lhs' type is fixed and the rhs may or may not be explicitely coerced.

```suggestion
@[simp, norm_cast] lemma coe_sum (s : Finset ι) (f : ι → α) :
    (∑ i in s, f i : WithTop α) = ∑ i in s, (f i : WithTop α) := map_sum addHom f s
```

Reply:
This pattern is already used elsewhere

Reply:
Fair, I'm just used to automatic coercion giving up when the lhs doesn't have an output type fixed